packages/tiny-graphql-query-compiler/spec/calls.spec.ts

```diff
@@ -0,0 +1,422 @@
+      type: "query",
+      fields: {
+        id: true,
+        name: true,
+        truncatedHTML: Call({ length: "$length" }),
+      },
+    });
+
+    expectValidGraphQLQuery(result);
+    expect(result).toContain(`"$length"`);
+    expect(result).toMatchInlineSnapshot(`
+      "query  {
+        id
+        name
+        truncatedHTML(length: "$length") 
+      }"
+    `);
+  });
+
+  test("it should compile a query that calls a field with null", () => {
+    const result = compile({
+      type: "query",
+      fields: {
+        id: true,
+        name: true,
+        truncatedHTML: Call({ length: null }),
+      },
+    });
+
+    expectValidGraphQLQuery(result);
+    expect(result).not.toContain("length:");
+    expect(result).toMatchInlineSnapshot(`
+      "query  {
+        id
+        name
+        truncatedHTML 
+      }"
+    `);
+  });
+
+  test("it should compile a query that calls a field with undefined", () => {
+    const result = compile({
+      type: "query",
+      fields: {
+        id: true,
+        name: true,
+        truncatedHTML: Call({ length: null }),
+      },
+    });
+
+    expectValidGraphQLQuery(result);
+    expect(result).not.toContain("length:");
+    expect(result).toMatchInlineSnapshot(`
+      "query  {
+        id
+        name
+        truncatedHTML 
+      }"
+    `);
+  });
+
+  test("it should compile a query that calls a field with a null variable", () => {
+    const result = compile({
+      type: "query",
+      fields: {
+        id: true,
+        name: true,
+        truncatedHTML: Call({ length: Var({ type: "String", value: null }) }),
+      },
+    });
+
+    expectValidGraphQLQuery(result);
+    expect(result).toMatchInlineSnapshot(`
+      "query ($length: String) {
+        id
+        name
+        truncatedHTML(length: $length) 
+      }"
+    `);
+  });
+
+  test("it should compile a query that calls a field with an undefined variable", () => {
+    const result = compile({
+      type: "query",
+      fields: {
+        id: true,
+        name: true,
+        truncatedHTML: Call({ length: Var({ type: "String", value: undefined }) }),
+      },
+    });
+
+    expectValidGraphQLQuery(result);
+    expect(result).toMatchInlineSnapshot(`
+      "query ($length: String) {
+        id
+        name
+        truncatedHTML(length: $length) 
+      }"
+    `);
+  });
+
+  test("it should compile a query that calls a field with a subselection", () => {
+    const result = compile({
+      type: "query",
+      fields: {
+        users: Call(
+          { first: 10 },
+          {
+            edges: {
+              node: {
+                id: true,
+              },
+            },
+          }
+        ),
+      },
+    });
+
+    expectValidGraphQLQuery(result);
+    expect(result).toMatchInlineSnapshot(`
+      "query  {
+        users(first: 10) { 
+          edges {
+            node {
+              id
+            }
+          } 
+        }
+      }"
+    `);
+  });
```

Comment:
A test I didn't see but I think would be good to have to clarify:

```js
compile({
  type: "query",
  fields: {
    something: Call({ length: Var({ type: "Int" }) }),
    another Call({ length: Var({ type: "Int!" }) }),
  },
});
```

So those are two different variables, but the argument name is the same. Do we throw? Does one clobber the other? Worth a test, eh?